quadfeather/tiler.py

```diff
@@ -82,6 +82,32 @@ def flatten_manifest(mani: TileManifest) -> List[Dict]:
     return d
 
 
+def rebatch(
+    input: Iterator[pa.RecordBatch], size: float, max_rows: float = float("inf")
+) -> Iterator[pa.Table]:
+    """
+    Assembles an iterator over record batches into a predictable memory size
+
+    Yields *more* than the requested size to avoid bad situations of zero-size arrays.
+    """
+    buffer = []
+    buffer_size = 0
+    buffer_rows = 0
+    for batch in input:
+        buffer_size += batch.nbytes
+        buffer_rows += batch.num_rows
+        buffer.append(batch)
+
+        if buffer_size > size or buffer_rows > max_rows:
+            yield pa.Table.from_batches(buffer).combine_chunks()
+            buffer = []
```

Comment:
Super nit: Not sure how big these chunks are getting but may as well create the table, clear the buffer, then yield so you don't have extra buffer memory hanging around.

Reply:
I like this yeah you're right. willfix

Reply:
Fixed.